rosbag2_transport/src/rosbag2_transport/rosbag2_transport.cpp

```diff
@@ -97,7 +97,12 @@ void Rosbag2Transport::play(
 
     auto transport_node = setup_node(play_options.node_prefix);
 
-    Player player(reader_, transport_node);
+    rosbag2_storage::MetadataIo metadata_io{};
+    rosbag2_storage::BagMetadata metadata{};
+    if (metadata_io.metadata_file_exists(storage_options.uri)) {
```

Comment:
Is this really OK not to have metadata?

Reply:
legacy rosbags

Reply:
ROS1 bags loaded with rosbag2_bag_v2 do not have metadata